lib/daru/io/link.rb

```diff
@@ -1,7 +1,9 @@
 module Daru
   class DataFrame
     class << self
-      def register_io_module(function, instance)
+      def register_io_module(function, instance=nil)
+        return define_singleton_method(function) { |*args| yield(*args) } if block_given?
```

Comment:
With explicit `&block` in args, you could've just `define_singleton_method(function, &block)` here.